From 604fdfd9153dc9ecd67f62105a022e7d13cf4e49 Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Tue, 7 Aug 2018 10:37:53 +0200
Subject: [PATCH] net: dsa: mv88e6xxx: fix write eeprom cmd

The MV88E6321 needs about 10 miliseconds to write the EEPROM command
register and to read back the ready flag. We have to wait for some ms
before doing the read operation to avoid MDIO bus IO errors.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 drivers/net/dsa/mv88e6xxx/global2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/dsa/mv88e6xxx/global2.c b/drivers/net/dsa/mv88e6xxx/global2.c
index cf686e7..c3d0fff 100644
--- a/drivers/net/dsa/mv88e6xxx/global2.c
+++ b/drivers/net/dsa/mv88e6xxx/global2.c
@@ -209,6 +209,8 @@ static int mv88e6xxx_g2_eeprom_cmd(struct mv88e6xxx_chip *chip, u16 cmd)
 	if (err)
 		return err;
 
+	usleep_range(10000, 15000);
+
 	return mv88e6xxx_g2_eeprom_wait(chip);
 }
 
-- 
2.7.4

